program-runtime/src/invoke_context.rs

```diff
@@ -847,50 +847,38 @@ pub struct MockInvokeContextPreparation {
 pub fn prepare_mock_invoke_context(
     program_indices: &[usize],
     instruction_data: &[u8],
-    keyed_accounts: &[(bool, bool, Pubkey, Rc<RefCell<AccountSharedData>>)],
+    transaction_accounts: Vec<(Pubkey, AccountSharedData)>,
```

Comment:
Is this same paradign going to be extended beyond mock?  If so, are we going to be allocating a lot of vectors as we pass this information around?

Reply:
No, the `Vec` allocation only happens this way for the tests / mocks, and only for now, as it might be refined to become index based as well (see comment above). The rest of the program-runtime will be directly migrated to the new structs defined in #21706 and they work reference and index based so there should not be more allocation happening than what we had until now.